### DISCUSSION TITLE
prisma update not triggered when there are no await inside a loop

### DISCUSSION THREAD

**Discussion by alan345** [prisma/prisma#3480] [Q&A] [answered]
this code dosen't work. users have not been updated: 

```ts
    for (const user of users) {
        ctx.prisma.user.update({ where: { id: user.id }, data: { mailSent: true } })
    }
```
this code works. users have been updated.

```ts
    for (const user of users) {
        await ctx.prisma.user.update({ where: { id: user.id }, data: { mailSent: true } })
    }
```

No error is dropped.

**Comment by ryands17** [comment, marked as answer]
Hey @alan345 👋 

Yes this is because the calls made to Prisma client are lazy. They are only evaluated when await is called, which is why the first one doesn't work. You can also read more about this [here](https://github.com/prisma/docs/issues/800).

**Reply by alan345** [reply]
Thank for the clarification!

**Reply by austonpramodh** [reply]
@ryands17 This seems really interesting. I am interested to know how does the Prisma api knows if await is called or not?

Sorry if it's a silly question, I wasn't able to find it on google. Searching through the Prisma repository seemed overkill.

Thank you.

**Reply by ryands17** [reply]
@austonpramodh 👋 

It uses something like this: https://github.com/sindresorhus/p-lazy

**Reply by austonpramodh** [reply]
Interesting. Thank you @ryands17

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
